kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/MaterializedRowTrackingColumn.java

```diff
@@ -145,6 +148,39 @@ public static Optional<Metadata> assignMaterializedColumnNamesIfNeeded(Metadata
         : Optional.of(metadata.withMergedConfiguration(configsToAdd));
   }
 
+  /**
+   * Converts a logical column to its corresponding physical column if it is a row tracking metadata
+   * column (ROW_ID or ROW_COMMIT_VERSION).
+   *
+   * <p>Note that we must not mark the physical columns as metadata columns because as far as the
+   * parquet reader is concerned, these columns are not metadata columns.
+   *
+   * @param column The logical column to convert.
+   * @param metadata The current metadata of the table.
+   * @return An Optional containing the physical column if conversion is applicable;
+   *     Optional.empty() otherwise.
+   */
+  public static Optional<StructField> convertToPhysicalColumn(
+      StructField column, Metadata metadata) {
+    if (column.equals(METADATA_ROW_ID_COLUMN)) {
+      return Optional.of(
+          new StructField(
+              metadata.getConfiguration().get(ROW_ID.getMaterializedColumnNameProperty()),
+              column.getDataType(),
+              column.isNullable()));
```

Comment:
The materialized fields should always be nullable. E.p. the current metadata struct fields are defined as non-nullable (which isn't actually correct yet but will be once we populate default values), but we do expect nulls from the materialized column.